developers/weaviate/current/configuration/authentication.md

```diff
@@ -134,17 +123,55 @@ services:
       AUTHENTICATION_OIDC_SKIP_CLIENT_ID_CHECK: 'false'
 ```
 
+#### Weaviate OpenID endpoint
+
+If you have authentication enabled, you can request all details needed from the following endpoint:
+
+```bash
+$ curl http://localhost:8080/v1/.well-known/openid-configuration
+```
+
 ### How to use
 
-1. Obtain a valid token from the token issuer you configured.
-2. Send this token along any REST request in the Header like so: `Authorization: Bearer <token>`. Make sure to replace `<token>` with your actual token.
+OIDC authentication requires obtaining a valid token from the token issuer so that it can be sent in the header of any REST request.
+
+The OIDC standard allows for many different methods *(flows)* of obtaining a token. The appropriate method can vary depending on your situation, including configurations at the token issuer, and your requirements. 
+
+While it is outside the scope of our documentation to cover every OIDC authentication flow, some possible options are to:
+- Use the `resource owner password flow` for trusted applications. 
+- Or use the `hybrid flow` if Microsoft Azure AD is your token issuer or if you would like to prevent exposing passwords.
+
+We outline the steps below for both methods of obtaining a token.
+
+#### Resource owner password flow
+1. Send a GET request to `[WEAVIATE_URL]/v1/.well-known/openid-configuration` to fetch Weaviate's OIDC configuration (`wv_oidc_config`)
+2. Parse the `clientId` and `href` from `wv_oidc_config`
+3. Send a GET request to `href` to fetch the token issuer's OIDC configuration (`token_oidc_config`)
+4. Ensure that `password` is in the `grant_types_supported` list in `token_oidc_config`.
```

Comment:
AFAIK azure does not have the `grant_types_supported` in their response (but I'm still not sure if they support user/PW at all)

Reply:
Right. I see that the key is optional (https://openid.net/specs/openid-connect-discovery-1_0.html). I will edit to say something about checking for support instead.

Reply:
Wrote "If `token_oidc_config` includes the optional `grant_types_supported` key, check that `password` is in the list of values."

Reply:
In the following sentence you write "If this is not the case, the token issuer is likely not configured for resource owner password flow. You may need to reconfigure the token issuer or use another method."
It might be unclear what "this" refers to, eg if "pw not in grant_types_supported" or "grant_types_supported in token_oidc_config" (or both). Could you clarify?

Reply:
Yup understood. Does the edited text make sense?

Reply:
Yes  👍